### PR TITLE
Lakka5x:Exclude yabasanshiro core on GPICase/ Pi02GPi device building

### PR DESCRIPTION
**Pull request by ShigeakiAsai**
# Pull requests overview

This pull request is for issue #1880.

Dear Lakka team

I would like to suggest about this issue that the yabasanshiro core is excluded on building GPICase and Pi02GPi devices.

The reason is bellow.

In GPICase and Pi02GPi device, both use OPENGLES="bcm2835-driver"in device options file.

- Lakka-LibreELEC/projects/RPi/devices/GPICase/options
- Lakka-LibreELEC/projects/RPi/devices/Pi02GPi/options

bcm2835-driver copies related header files to toolchain during building.
But GLES 3.0 header files are not included in these header files.
Because bcm2835 supports only GLES 2.0 I think.

On the other hand, the yabasanshiro core refers GLES 3.0 header files.

Therefore, GPICase and Pi02GPi building are failed.

By the way, RPi.arm device excludes the yabasanshiro core.

Just like RPi.arm device, it's better to exclude yabasanshiro core on both devices I think.

# Pull requests information

Modified file is 1 file only.
distributions/Lakka/options

Explain using comitted source file.

line :
322 Separate "PROJECT" and "DEVICE". And "${PROJECT}" = "RPi" block start.
323   Case of "${DEVICE}" = "RPi" or "${DEVICE}" = "GPICase" block. (Current)
.
.
378-380 Adding for the yabasanshiro core is excluded on building GPICase and Pi02GPi devices.
 No use "elif" in here, because already used "GPICase" and "Pi02GPi" as if expression.
381  "${PROJECT}" = "RPi" block end, And replace "if" to "elif".

# Local test result with this pull request
- DISTRO=Lakka PROJECT=RPi DEVICE=GPICase ARCH=arm make image
->Passed
- DISTRO=Lakka PROJECT=RPi DEVICE=Pi02GPi ARCH=arm make image
->Failed at ppsspp core. (I think this fail is caused by another reason.)
- DISTRO=Lakka PROJECT=RPi DEVICE=RPi ARCH=arm make image
->Passed
- DISTRO=Lakka PROJECT=RPi DEVICE=RPi4 ARCH=aarch64 make image
->Passed
- DISTRO=Lakka PROJECT=Generic ARCH=x86_64 make image
->Passed

And I test GPi CASE with Raspberry Pi ZERO W using this img.
Boot starts, logo appered, but don'd chage graphical menu screen.
I will investigate this.

Thanks
ASAI, Shigeaki

### CODE REVIEW

**Comment by ShigeakiAsai** [comment]
> And I test GPi CASE with Raspberry Pi ZERO W using this img.
> Boot starts, logo appered, but don'd chage graphical menu screen.

I'm waiting more time, MAIN MENU screen is appeared.
(Before MAIN MENU screen is changed , some pinctrl-bcm2835 message appears.)






Sorry for my mis-report.

ASAI, Shigeaki

File: distributions/Lakka/options [hunk 1]
```diff
@@ -319,7 +319,8 @@
                  "
 
 # disable some cores
-    if [ "${PROJECT}" = "RPi" -a "${DEVICE}" = "RPi" -o "${DEVICE}" = "GPICase" ]; then
+  if [ "${PROJECT}" = "RPi" ]; then
+    if [ "${DEVICE}" = "RPi" -o "${DEVICE}" = "GPICase" ]; then
```
Comment (Ntemis) [inline distributions/Lakka/options]: Why you are touching these? They have nothing to do with your commit message

File: distributions/Lakka/options [hunk 2]
```diff
@@ -374,7 +375,10 @@
     elif [ "${DEVICE}" = "Pi02GPi" ]; then
       EXCLUDE_LIBRETRO_CORES+=" kronos openlara play swanstation"
     fi
-  if [ "${PROJECT}" = "Amlogic" -o "${PROJECT}" = "Rockchip" -o "${PROJECT}" = "Allwinner" ]; then
+    if [ "${DEVICE}" = "GPICase" -o "${DEVICE}" = "Pi02GPi" ]; then
```
Comment (Ntemis) [inline distributions/Lakka/options]: You are changing too many variables here, please just do an exclude case and leave the rest changes unharmed

**Review by Ntemis — CHANGES_REQUESTED** [review]
Not ready to be merged

File: distributions/Lakka/options (hunk 1 above)
Comment (ShigeakiAsai) [reply distributions/Lakka/options]: There are 2 reasons.
(1) I know you changed it in last commit, but it is may not good compared to more old one.
I mean " if ( A and B or C ) " is may not judged correctly.
- ( A anb B ) or ( A and C ) ?
- ( A and B ) or ( C ) ?

(2) Please refer to all of distributions/Lakka/options file Line 372/373 and 374/375.
372/373 elif [ "${DEVICE}" = "RPi2" ]; then
374/375 elif [ "${DEVICE}" = "Pi02GPi" ]; then
These "elif" have not <"${PROJECT}" = "RPi" -a> in currently.

Therefore I restored it.

File: distributions/Lakka/options (hunk 2 above)
Comment (ShigeakiAsai) [reply distributions/Lakka/options]: I want to add 3 lines before Line 377
`
    if [ "${DEVICE}" = "GPICase" -o "${DEVICE}" = "Pi02GPi" ]; then
      EXCLUDE_LIBRETRO_CORES+=" yabasanshiro"
    fi
`

And so, I want to change "if" to "elif" at line 377.

File: distributions/Lakka/options (hunk 1 above)
Comment (Ntemis) [reply distributions/Lakka/options]: A=PROJECT  B or C is DEVICE so if a and if b or c and is perfectly fine as it is

Comment (ShigeakiAsai) [reply distributions/Lakka/options]: I've got it

**Comment by ShigeakiAsai** [comment]
@Ntemis 
I will replace (change modification ) this pull request to small changing.
Could you please wait a moment?

ASAI, Shigeaki

**Comment by ShigeakiAsai** [comment]
Could you please replace this PullRequest with new commit.
261427bc09a572438546735179c38bd4dc4e6229


Test for this modification
DISTRO=Lakka PROJECT=RPi DEVICE=Pi02GPi ARCH=arm make image
->Passed (*1)(*2)(*3)
DISTRO=Lakka PROJECT=RPi DEVICE=GPICase ARCH=arm make image
->Passed (*2)

*1 Another build fail is occured in PPSSPP core.
   This is existing fail that has been occurring at Sep 14.
   I will report about this fail via new issue.

*2 New build fail is occured in Retroarch.
   This is new build fail that has been occurring at Oct 1.
   I used the previous commit for confirmation this PullRequest.
   packages/lakka/retroarch_base/retroarch/package.mk
   Now I am investigating this.

*3 New build fail is occured in mame core.
   This is new build fail that has been occurring at Oct 1.
   I used the previous commit for confirmation this PullRequest.
   packages/lakka/retroarch_base/retroarch/package.mk
   I will investigate this after *2.


Test for regression
DISTRO=Lakka PROJECT=RPi DEVICE=RPi ARCH=arm make image
->Passed
DISTRO=Lakka PROJECT=RPi DEVICE=RPi4 ARCH=aarch64 make image
->Passed


Thanks
ASAI Shigeaki from Japan

File: distributions/Lakka/options (hunk 2 above)
Comment (ShigeakiAsai) [reply distributions/Lakka/options]: Update

**Comment by ShigeakiAsai** [comment]
I think this Pull Request was something noisy.
I close this Pull Request with comment, and I hope to modify by Lakka team please.

Thanks,
ASAI, Shigeaki

**Pull request closed without merging**